src/main.rs

```diff
@@ -234,37 +251,49 @@ async fn handle_request(
         }
     };
 
-    let bytes = match hyper::body::to_bytes(body).await {
-        Ok(body) => body.to_vec(),
+    let p = path_name(&path);
+
+    let header_sender = match ratelimiter.ticket(path).await {
+        Ok(sender) => sender,
         Err(e) => {
-            error!("Failed to receive incoming request body: {:?}", e);
-            return Err(RequestError::ChunkingRequest { source: e });
+            error!("Failed to receive ticket for ratelimiting: {:?}", e);
+            return Err(RequestError::AcquiringTicket { source: e });
         }
     };
 
-    let path_and_query = if let Some(v) = uri.path_and_query() {
-        v.as_str().replace(&api_url, "")
-    } else {
-        error!("No path in URI: {:?}", uri);
-        return Err(RequestError::NoPath { uri });
+    request.headers_mut().insert(
+        AUTHORIZATION,
+        HeaderValue::from_str(&token).expect("token is valid ascii"),
```

Comment:
This should arguably be an error since not all UTF-8 is valid ASCII (panic = potential DOS?)